plugins/dac_data_manager.c

```diff
@@ -141,16 +141,6 @@ static bool tx_channels_check_valid_setup(struct dac_buffer *dbuf);
 
 static const gdouble abs_mhz_scale = -1000000.0;
 static const gdouble khz_scale = 1000.0;
-static const char *default_channel_names[8] = { // TO DO: Should generate these automatically if number of TX will be variable
```

Comment:
this patch adds a lot of allocation & free-ing;
which just complicates things [for both review & future bug findings];  

iio-oscilloscope wasn't particularly a stable piece of software, and this doesn't help to get close to that point

maybe, let's change things a bit; 
either:
1. we can we build/alloc this list on a single init/constructor path, and re-use the `default_channel_names` var-name ; and destroy the list of the destructor path
2.  paths can be constructed inside a buffer that is passed to the `build_` function

Reply:
I understand what you mean but I'm not using this list and there is no need to reuse the information inside that list (at the moment). The default names are used only in a particular initialization step and that is it. If this changes then yes, your suggestion should be the right approach.

Reply:
following comment https://github.com/analogdevicesinc/iio-oscilloscope/pull/133/commits/d5e9b4da94f4fdbe8145685cd75beb4b6a5dabff#r259751705  , this could e converted into `static const char **default_channel_names ` and `build_default_channel_name_from_index()` constructs this list on the first call (i.e. if `default_channel_names` is NULL)

in that case all that needs to be remembered, is to de-alloc `default_channel_names` [it could be renamed to something else] when the module is destroyed or unloaded

Reply:
The list cannot be static, it needs to belong to an instance. You could have many instances of dac_data_manager (simultaneously), each with its own number of channels. E.g one instance with 2 channels and one instance with 64 channels.